### PR TITLE
[#143] added wifi image and duplicated ping rover function, working on modifiyng according

### LINKED ISSUE #143: Show latency/ping through an icon
Show the ping through an icon. Similar to what you see in most games. It would auto refresh every 4-5 seconds to display an accurate ping. This is an improvement over having to click each device to see if it pings or not through the chat.

~OBC, ROS and Embedded tags are included because pinging checks latency to either the Odroid (OBC) or a microcontroller connected to the Odroid.~

### PR DESCRIPTION
**Pull request by JonathanSolvesProblems**
# Assignee Section

## Description
Added wifi image in top left of gui in the navbar.html template, which inherits over all other html pages. In addition, made a duplicate of the ping rover function. Need to fix that.

[If applicable, include a list of main points]
- Example bug fixed.
- Added example function with the responsibility of example.

### Steps for Testing
[Write all the steps necessary to test the feature(s)]
1) Run ./app.py in the basestation folder
2) Type localhost:5000 on an internet page
3) The GUI will load and a wifi icon in the top left will appear

closes #143

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [ ] Local Test Performed Successfully
- [ ] [General Integration Test](https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit) Performed Successfully

For Pull Requests that do not include code changes, it is not required to perform the tests above.

### CODE REVIEW

File: robot/basestation/static/img/connection_icons/connected_bar.desktop [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Link to Full Wi-Fi Signal
+Type=Link
+URL=https://cdn.pc.net/images/db/wi-fi_signal_full.png
+Icon=image-png
+Name[en_US]=connected_bar
```
Comment (MartensCedric) [inline robot/basestation/static/img/connection_icons/connected_bar.desktop]: do we need the image in .desktop?

File: robot/basestation/static/img/connection_icons/not_connected_bar.desktop [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Link to wifi_bang
+Type=Link
+URL=https://2xom5j4aqh9o3quhfzkge6t1-wpengine.netdna-ssl.com/wp-content/uploads/2015/02/wifi_bang-300x300.png
+Icon=image-png
```
Comment (MartensCedric) [inline robot/basestation/static/img/connection_icons/not_connected_bar.desktop]: same comment as above

File: robot/basestation/static/js/elements/feed/connection-feed.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+$(document).ready(() => {
+
+	var connected = 
+
+})
```
Comment (MartensCedric) [inline robot/basestation/static/js/elements/feed/connection-feed.js]: what is this.

Please remove this file lol

File: robot/basestation/templates/elements/feed/ConnectionFeed.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+<head>
+	<title>Connection Feed</title>
+</head>
+<body>
+
+<img src="home/jonathansolvesproblems/Programming/robotics-prototype/robot/basestation/static/img/connected_bar">
+
+<img src="home/jonathansolvesproblems/Programming/robotics-prototype/robot/basestation/static/img/not_connected_bar" css="display:none">
```
Comment (MartensCedric) [inline robot/basestation/templates/elements/feed/ConnectionFeed.html]: Jonathan solves problems

Comment (JonathanSolvesProblems) [reply robot/basestation/templates/elements/feed/ConnectionFeed.html]: removed ConnectionFeed.html

File: robot/basestation/static/js/elements/feed/connection-feed.js (hunk 3 above)
Comment (JonathanSolvesProblems) [reply robot/basestation/static/js/elements/feed/connection-feed.js]: removed

File: robot/basestation/static/img/connection_icons/not_connected_bar.desktop (hunk 2 above)
Comment (JonathanSolvesProblems) [reply robot/basestation/static/img/connection_icons/not_connected_bar.desktop]: removed connectFolder with images, not being used

File: robot/basestation/static/img/connection_icons/connected_bar.desktop (hunk 1 above)
Comment (JonathanSolvesProblems) [reply robot/basestation/static/img/connection_icons/connected_bar.desktop]: removed connectFolder with images, not being used

File: robot/basestation/static/js/app.js [hunk 5]
```diff
@@ -254,9 +254,55 @@ function pingOdroid (timeoutVal = REQUEST_TIMEOUT) {
   lastCmdSent = new Date().getTime()
 }
 
+<<<<<<< HEAD
```
Comment (MartensCedric) [inline robot/basestation/static/js/app.js]: I think you are not done with a merge

**Comment by MartensCedric** [comment]
Also your PR doesnt at all close #143, it seems like you just made a section

**Comment by MartensCedric** [comment]
@JonathanSolvesProblems make sure you fix that merge leftover so we can review your PR again

File: robot/basestation/static/js/app.js [hunk 6]
```diff
@@ -254,9 +254,55 @@ function pingOdroid (timeoutVal = REQUEST_TIMEOUT) {
...
+
+// pingOBC(timeout, callback)
+// call the callback with the ping and if its successful or not
+
+// will display the highest function.
+let HIGH = function() {
+
+}
```
Comment (MartensCedric) [inline robot/basestation/static/js/app.js:265]: very useful

File: robot/basestation/static/js/app.js [hunk 7]
```diff
@@ -254,9 +254,55 @@ function pingOdroid (timeoutVal = REQUEST_TIMEOUT) {
...
+
+// 100ms, or ma for nothing
+=======
 /*
 plays an error sound once -> warning will not play if the document has not been interacted with yet
 */
 function errorSound (){
   new Audio(ERROR_SOUND_PATH).play()
 }
+>>>>>>> master
```
Comment (MartensCedric) [inline robot/basestation/static/js/app.js:308]: please remove this shit

**Comment by JonathanSolvesProblems** [comment]
> image
> There's errors in the JS

Resolved, webpage runs as suppose to. There were errors with the merge conflicts.

**Comment by JonathanSolvesProblems** [comment]
> @JonathanSolvesProblems make sure you fix that merge leftover so we can review your PR again

merge leftover fixed, working on improving the aesthetics and overall functionality.

**Comment by JonathanSolvesProblems** [comment]
> @JonathanSolvesProblems make sure you fix that merge leftover so we can review your PR again

File: robot/basestation/static/js/app.js (hunk 5 above)
Comment (JonathanSolvesProblems) [reply robot/basestation/static/js/app.js]: > I think you are not done with a merge

resolved

**Comment by JonathanSolvesProblems** [comment]
> app.js definitely doesnt work. Also you removed files in docs/media, idk if this was by accident or not

app.js now works. Not sure which foles in docs/media. Will look into.

**Comment by JonathanSolvesProblems** [comment]
> You also uploaded a .swp file. I dont see how this should work lol, plus your paths are obviously invalid outside your computer.

paths fixed.

**Pull request closed without merging**